File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const n = speeds.length; 
    const days = progresses.map((item, i) => Math.ceil((100 - item) / speeds[i]))
    const answer = [];
    console.log(days);
    
    let count = 0;
    let maxDay = days[0];
    for(let i = 0; i < n; i++){
        if(days[i] <= maxDay){
            count++
        } else {
            answer.push(count);
            count = 1;
            maxDay = days[i];
        }
    }
    answer.push(count);
    return answer
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
import math

def solution(progresses, speeds):
    days = [0] * len(progresses)
    queue = deque()
    deploys = []
    for i in range(len(progresses)):
        days[i] = math.ceil((100 - progresses[i]) / speeds[i])
    print(days)
    front = days[0]
    queue.append(front)
    
    for i in range(1, len(days)):
        if front < days[i]:
            count = 0
            while len(queue) > 0:
                queue.popleft()
                count += 1
            deploys.append(count)
            front = days[i]
                
        queue.append(days[i])
    
    if len(queue) > 0:
        deploys.append(len(queue))
        
    return deploys